tests/test_dataset_utils.py

```diff
@@ -192,6 +196,37 @@ def test_is_noop_mapping_false():
         assert not is_noop_mapping(df)
 
 
+def test_remove_invalid_null_timestamps():
+    spark = get_spark_session()
+    df = spark.createDataFrame(
+        [
+            # No nulls
+            {"timestamp": 1, "county": "Jefferson", "subsector": "warehouse", "value": 4},
+            {"timestamp": 2, "county": "Jefferson", "subsector": "warehouse", "value": 5},
+            # Nulls and valid values
+            {"timestamp": None, "county": "Boulder", "subsector": "large_office", "value": 0},
+            {"timestamp": 1, "county": "Boulder", "subsector": "large_office", "value": 4},
+            {"timestamp": 2, "county": "Boulder", "subsector": "large_office", "value": 5},
+            # Only nulls
+            {"timestamp": None, "county": "Adams", "subsector": "retail_stripmall", "value": 0},
+            {"timestamp": None, "county": "Denver", "subsector": "hospital", "value": 0},
+        ]
+    )
+    stacked = ["county", "subsector"]
+    time_col = "timestamp"
+    result = remove_invalid_null_timestamps(df, {time_col}, stacked)
+    result = (
+        df.join(
+            df.groupBy(*stacked).agg(F.count_distinct(time_col).alias("count_time")), on=stacked
+        )
+        .filter(f"{time_col} IS NOT NULL or count_time == 0")
+        .drop("count_time")
+    )
```

Comment:
Why is result overwritten with this line that duplicates the logic in the function `remove_invalid_null_timestamps`?

Reply:
Oops.